### PR TITLE
fix CombineTransposeConstBinary

### PR DESCRIPTION
**Pull request by xhuohai**
constrain condition of CombineTransposeConstBinary and register to TargetIndependentPass.

### CODE REVIEW

File: src/Nncase.Transform/Rules/Neutral/CombineTranspose.cs [hunk 1]
```diff
@@ -61,7 +61,7 @@ public sealed partial class CombineTransposeConstBinary : IRewriteRule
     public CombineTransposeConstBinary()
     {
         var perm = IsWildcard("perm");
-        Pattern = IsAlt(IsBinary("binary", _ => true, IsTranspose(IsWildcard("x"), perm), IsConst("y")), IsBinary("binary", _ => true, IsConst("x"), IsTranspose(IsWildcard("y"), perm)));
+        Pattern = IsAlt(IsBinary("binary", _ => true, IsTranspose(IsWildcard("x"), perm), IsConst("y", y => y.CheckedShape.Count == 1)), IsBinary("binary", _ => true, IsConst("x", x => x.CheckedShape.Count == 1), IsTranspose(IsWildcard("y"), perm)));
```
Comment (sunnycase) [inline src/Nncase.Transform/Rules/Neutral/CombineTranspose.cs:64]: It's more semantically meaningful to use `IsConst("y") with { TypePattern = HasRank(1) }`.

**Review by FusionBolt — CHANGES_REQUESTED** [review]
should add tests for match and mismatch

File: src/Nncase.Transform/Rules/Neutral/CombineTranspose.cs (hunk 1 above)
Comment (xhuohai) [reply src/Nncase.Transform/Rules/Neutral/CombineTranspose.cs]: resolved

**Comment by xhuohai** [comment]
> should add tests for match and mismatch

added.

**Review by FusionBolt — APPROVED** [review]

**Review by sunnycase — APPROVED** [review]

**Pull request merged**